flask_resources/errors.py

```diff
@@ -72,50 +79,73 @@ def get_body(self, environ=None):
+        return json.dumps(body)
+
 
+class SchemaValidationError(HTTPJSONException):
+    """Marshmallow schema validation exception."""
+
+    def __init__(self, errors=None, **kwargs):
+        """Initialize exception."""
+        _errors = []
+        if errors:
+            for field, messages in errors.items():
+                _errors.extend([_FieldError(field, msg) for msg in messages])
+        super(SchemaValidationError, self).__init__(errors=_errors, **kwargs)
 
-# FIXME: From here down need review
+    class _FieldError(object):
+        """Represents a field level error.
 
+        .. note:: This is not an actual exception.
+        """
 
-class FieldError(object):
-    """Represents a field level error.
+        def __init__(self, field, message, code=None):
+            """Init object.
 
-    .. note:: This is not an actual exception.
-    """
+            :param field: Field name.
+            :param message: The text message to show.
+            :param code: The HTTP status to return. (Default: ``None``)
+            """
+            self.res = dict(field=field, message=message)
+            if code:
+                self.res["code"] = code
 
-    def __init__(self, field, message, code=None):
-        """Init object.
+        def to_dict(self):
+            """Convert to dictionary.
 
-        :param field: Field name.
-        :param message: The text message to show.
-        :param code: The HTTP status to return. (Default: ``None``)
-        """
-        self.res = dict(field=field, message=message)
-        if code:
-            self.res["code"] = code
+            :returns: A dictionary with field, message and, if initialized, the
+                HTTP status code.
+            """
+            return self.res
 
-    def to_dict(self):
-        """Convert to dictionary.
+    def get_errors(self):
+        """Get errors.
 
-        :returns: A dictionary with field, message and, if initialized, the
-            HTTP status code.
+        :returns: A list containing a dictionary representing the errors.
         """
-        return self.res
+        return [e.to_dict() for e in self.errors] if self.errors else None
 
 
-#
-# Search
-#
-class SearchPaginationRESTError(RESTException):
-    """Search pagination error."""
+class BadRequestRESTError(HTTPJSONException):
+    """Bad request related exceptions."""
 
     code = 400
 
-    def __init__(self, errors=None, **kwargs):
-        """Initialize exception."""
-        _errors = []
-        if errors:
-            for field, messages in errors.items():
-                _errors.extend([FieldError(field, msg) for msg in messages])
-        super(SearchPaginationRESTError, self).__init__(errors=_errors, **kwargs)
+
+class GoneRESTError(HTTPJSONException):
+    """Gone related exceptions."""
+
+    code = 410
+
+
+class ForbiddenRESTError(HTTPJSONException):
+    """Forbidden related exceptions."""
+
+    code = 403
+
+
+class UnsupportedMediaTypeRESTError(HTTPJSONException):
```

Comment:
I think the message should be constructed similar to https://github.com/inveniosoftware/invenio-rest/blob/55d8d325feb792ecd663f49459c8431c3a4b320b/invenio_rest/errors.py#L89